### DISCUSSION TITLE
setup Prop withDefaults wrong out module

### DISCUSSION THREAD

**Discussion by ashuicoder** [vuejs/rfcs#381] [RFC Discussions]
If define interface `IProps ` in `script`，it succussful，but defined out module is wrong `TypeError: Cannot read property 'content' of null`
```ts
import { withDefaults, defineProps } from 'vue'
import { IProps } from './type'

const props = withDefaults(defineProps<IProps>(), {
  height: '200px'
})
```

`type.ts`
```ts
export interface IProps {
  // loading: boolean
  // error: boolean
  height?: string
}

```

**Comment by JessicaSachs** [comment]
I believe that `defineProps` and `withDefaults` are compiler hints that will only work inside of SFC files. What version of Vue are you using? (I'm also not sure if this repo is the place for this discussion).

**Reply by ashuicoder** [reply]
vue version 3.2.3

**Comment by posva** [comment]
See https://github.com/vuejs/vue-next/issues/4294
